resources/lib/videoplayer.py

```diff
@@ -56,22 +60,35 @@ def start_playback(self):
         self._prepare_and_start_playback()
 
         # wait with starting the threads until playback has started for guarantee, otherwise they will immediately exit
-        self._handle_skipping()
-        self._handle_update_playhead()
 
-    def is_playing(self) -> bool:
+    def isPlaying(self) -> bool:
+        if not self._stream_data:
+            return False
+        if self._player.isPlaying() and self._stream_data.stream_url == self._player.getPlayingFile():
+                return True
+        else:
+            return False
+
+    def isStartingOrPlaying(self) -> bool:
         """ Returns true if playback is running. Note that it also returns true when paused. """
 
         if not self._stream_data:
             return False
 
-        if not self._player.isPlaying():
-            return False
+        if self.isPlaying():
+            self.waitForStart = False
+            return True
 
-        return self._stream_data.stream_url == self._player.getPlayingFile()
+        # Wait max 5 sec for start playing the stream
+        if (time.time() - self.createTime) > 5:
```

Comment:
Could be less than 5 seconds, since `createTime` is set when creating the instance. When starting playback it first has to collect the data for the stream, which takes a second or so. But should not be a biggie.

What bugs me more is that the way I've checked if the playback has started using `isPlaying` doesn't seem to work in all cases and thus we have to rely on a timer here - timers are not reliable as well, but I guess there is no other way.

Reply:
`isPlaying` works in "all" cases, but it takes some time for it to return it correctly.

The 5 seconds should make no difference here.
The goal of the timer is that we don't get stuck in the loop forever, i.e. a timeout. 
We can increase the time, e.g. to 20 seconds. For me it takes about 1-3 seconds until the player reports back stream started.
Theoretically, we can remove the time completely and just wait forever.
But if we reach the maximum time and Kodi does not report back, something has gone wrong.
We can issue an error here.